### ISSUE TITLE
Temporary dict does not merge (overwrite) duplicated keys, resulting passing the same keyword multiple times. 

### ISSUE THREAD

**Issue by yamada-github-account** [cython/cython#3756]
The following code works fine at Python (CPython), but get error at Cython (v0.29.21) with Python 3.6.9.

```python
def print_dict(**kwargs):
    print(kwargs)

def call():
    d1 = {"1": 1, "2": 2}
    d2 = {"1": -1, "3": -3}

    # This is OK
    cdef dict d = {**d1,**d2}
    # cdef d = {**d1,**d2}  # OK, too
    # d = {**d1,**d2} # OK, too
    print_dict(**d)

    # This got error -> TypeError: function() got multiple values for keyword argument '1'
    print_dict(**{**d1,**d2})
```

It seems that cython version does not merge (overwrite) duplicated keys of temporary `dict` before unpacked by `**` operator.
Is this an intentional behavior or bug?

This code itself does not need cython features, but this kind of code is used inside my cython (.pyx) code.

**Comment by da-woods** [comment]
It works for me on the current master branch but not on the latest 0.29 release.

This looks like a bug. It looks like it's getting (incorrectly) converted to `print_dict(**d1, **d2)`.

**Comment by da-woods** [comment]
Relevant commit is 0919f3e12f9e63add4b063d4ddb67ef0165e7ff9

**Comment by yamada-github-account** [comment]
@da-woods 
Thanks for your reply.

I tested by myself and confirmed it is fixed at the master branch of version 3.0a5.

It seems this issue is duplicated with #2963, and will be fixed at the next major version 3.0, so that I close this issue.

Thank you for your support.

**Comment by scoder** [comment]
> It seems this issue is duplicated with #2963, and will be fixed at the
>next major version 3.0, so that I close this issue.

It looks like something that we could backport to 0.29.x, though.
